src/core/protocol-handlers/github-handler.ts

```diff
@@ -0,0 +1,404 @@
+      } else {
+        // Direct file access
+        const filePath = parts.slice(2).join('/');
+        return await this.loadDirectFile(owner, repo, filePath, options);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        if (error.name === 'AbortError') {
+          throw new Error(`Request timeout: Failed to load calendar from GitHub: ${location}`);
+        }
+        Logger.error(`Failed to load calendar from GitHub: ${location}`, error);
+        throw error;
+      } else {
+        const errorMessage = `Unknown error loading calendar from GitHub: ${location}`;
+        Logger.error(errorMessage);
+        throw new Error(errorMessage);
+      }
+    }
+  }
+
+  /**
+   * Load calendar from repository index
+   */
+  private async loadFromRepositoryIndex(
+    owner: string,
+    repo: string,
+    calendarId?: string,
+    options: LoadCalendarOptions = {}
+  ): Promise<SeasonsStarsCalendar> {
+    // Load the repository index
+    const index = await this.loadRepositoryIndex(owner, repo, options);
+
+    // Use universal selection logic
+    const selectionResult = selectCalendarFromIndex(
+      index,
+      calendarId,
+      `Repository ${owner}/${repo}`
+    );
+
+    if (selectionResult.error) {
+      throw new Error(selectionResult.error);
+    }
+
+    if (selectionResult.selectedEntry) {
+      Logger.debug(
+        `Loading calendar from index entry: ${selectionResult.selectedEntry.id} -> ${selectionResult.selectedEntry.file}`
+      );
+      return await this.loadDirectFile(owner, repo, selectionResult.selectedEntry.file, options);
+    }
+
+    throw new Error(`Unexpected error in calendar selection from index`);
+  }
+
+  /**
+   * Load repository index file
+   */
+  private async loadRepositoryIndex(
+    owner: string,
+    repo: string,
+    options: LoadCalendarOptions = {}
+  ): Promise<CalendarCollectionIndex> {
+    const apiUrl = `https://api.github.com/repos/${owner}/${repo}/contents/index.json`;
+
+    Logger.debug(`Loading repository index from GitHub: ${apiUrl}`);
+
+    const fetchOptions: RequestInit = {
+      method: 'GET',
+      headers: {
+        Accept: 'application/vnd.github.v3+json',
+        'User-Agent': 'FoundryVTT/Seasons-and-Stars',
+        ...options.headers,
+      },
+    };
+
+    if (options.timeout) {
+      const controller = new AbortController();
+      setTimeout(() => controller.abort(), options.timeout);
+      fetchOptions.signal = controller.signal;
+    }
+
+    const response = await fetch(apiUrl, fetchOptions);
+
+    if (!response.ok) {
+      if (response.status === 404) {
+        throw new Error(
+          `Repository index not found: ${owner}/${repo}/index.json. Repository may not have a calendar index or may not exist.`
+        );
+      } else if (response.status === 403) {
+        const rateLimitRemaining = response.headers.get('x-ratelimit-remaining');
```

Comment:
Is this a known or common header? Something github provides? Is it documented anywhere?